In [1]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

**Exploración inicial**

In [2]:
df_INAH_visitas = pd.read_csv('../data/raw/INAH_visitantes_museos_general.csv')
pd.set_option('display.max_columns', None)
df_INAH_visitas.head(5)

,anio,estado,clave_siinah,recinto,enero_nac,enero_ext,febrero_nac,febrero_ext,marzo_nac,marzo_ext,abril_nac,abril_ext,mayo_nac,mayo_ext,junio_nac,junio_ext,julio_nac,julio_ext,agosto_nac,agosto_ext,septiembre_nac,septiembre_ext,octubre_nac,octubre_ext,noviembre_nac,noviembre_ext,diciembre_nac,diciembre_ext
0,2025,Aguascalientes,2300100,Museo Regional de Historia de Aguascalientes,967,33,1721,27,1658,29,3726,61,2076,21,1378,50,1900,76,1571,46,1643,41,2959,66,3349,61,1660,61
1,2025,Baja California,2400100,"Museo Histórico Regional, Ex Cuartel de la Com...",1304,410,997,414,1932,292,1755,186,2544,409,3240,386,2100,511,2852,480,1649,132,3774,478,1784,516,1159,832
2,2025,Baja California Sur,2300200,Museo Regional de Antropología e Historia de B...,2266,983,1868,1101,2635,806,2337,544,2004,348,2009,547,2188,476,1040,567,940,346,2090,491,1969,515,1214,737
3,2025,Baja California Sur,2400200,Museo de Las Misiones Jesuíticas,659,179,887,474,798,551,1503,431,547,42,675,43,828,17,547,23,324,14,589,325,712,302,887,470
4,2025,Baja California Sur,2400300,Museo de las Pinturas Rupestres de San Ignacio,36,0,73,4,0,0,356,0,0,0,10,0,177,93,27,0,0,0,25,0,130,18,119,0


In [3]:
df_INAH_visitas.info()

<class 'pandas.DataFrame'>
RangeIndex: 3008 entries, 0 to 3007
Data columns (total 28 columns):
 #   Column          Non-Null Count  Dtype
---  ------          --------------  -----
 0   anio            3008 non-null   int64
 1   estado          3008 non-null   str  
 2   clave_siinah    3008 non-null   int64
 3   recinto         3008 non-null   str  
 4   enero_nac       3008 non-null   int64
 5   enero_ext       3008 non-null   int64
 6   febrero_nac     3008 non-null   int64
 7   febrero_ext     3008 non-null   int64
 8   marzo_nac       3008 non-null   int64
 9   marzo_ext       3008 non-null   int64
 10  abril_nac       3008 non-null   int64
 11  abril_ext       3008 non-null   int64
 12  mayo_nac        3008 non-null   int64
 13  mayo_ext        3008 non-null   int64
 14  junio_nac       3008 non-null   int64
 15  junio_ext       3008 non-null   int64
 16  julio_nac       3008 non-null   int64
 17  julio_ext       3008 non-null   int64
 18  agosto_nac      3008 non-null   int64
 

El conjunto de datos cuenta con 3008 registros/filas y 28 columnas(2 str y 26 int).

In [4]:
cantidad_nulos = df_INAH_visitas.isnull().sum()
print(f'Cantidad de valores nulos: {cantidad_nulos}')

Cantidad de valores nulos: anio              0
estado            0
clave_siinah      0
recinto           0
enero_nac         0
enero_ext         0
febrero_nac       0
febrero_ext       0
marzo_nac         0
marzo_ext         0
abril_nac         0
abril_ext         0
mayo_nac          0
mayo_ext          0
junio_nac         0
junio_ext         0
julio_nac         0
julio_ext         0
agosto_nac        0
agosto_ext        0
septiembre_nac    0
septiembre_ext    0
octubre_nac       0
octubre_ext       0
noviembre_nac     0
noviembre_ext     0
diciembre_nac     0
diciembre_ext     0
dtype: int64


In [5]:
cantidad_duplicados = df_INAH_visitas.duplicated().sum()
print(f'Cantidad de registros/filas duplicadas: \n{cantidad_duplicados}')

Cantidad de registros/filas duplicadas: 
0


No se identificaron valores nulos y duplicados.

In [6]:
annios = df_INAH_visitas['anio'].value_counts().sort_index(ascending=False)
print(annios)

anio
2025    104
2024    103
2023    101
2022    101
2021    101
2020    101
2019    101
2018    101
2017    101
2016     99
2015    102
2014    100
2013    100
2012    100
2011     99
2010     98
2009     99
2008    100
2007    101
2006     97
2005     98
2004     99
2003     99
2002    101
2001     95
2000     95
1999    103
1998    103
1997    103
1996    103
Name: count, dtype: int64


In [7]:
estados = df_INAH_visitas['estado'].value_counts().sort_index(ascending=True)
print(estados)

estado
Aguascalientes          30
Baja California         30
Baja California Sur    118
Campeche               120
Chiapas                120
Chihuahua               30
Ciudad de México       270
Coahuila                30
Colima                  30
Estado de México       236
Guanajuato             230
Guerrero                90
Hidalgo                136
Jalisco                120
Michoacán              150
Morelos                121
Nayarit                 60
Nuevo León              30
Oaxaca                 214
Puebla                 345
Querétaro               30
Quintana Roo            25
San Luis Potosí         30
Sinaloa                 28
Sonora                  30
Tabasco                 30
Tamaulipas              30
Tlaxcala                30
Veracruz               172
Yucatán                 63
Zacatecas               30
Name: count, dtype: int64


In [29]:
siinah = df_INAH_visitas['clave_siinah'].value_counts()
siinah

clave_siinah
4100015    39
2402000    38
4100016    37
2301000    31
4100017    31
           ..
2404000    23
2401600    20
4200004    10
4200002     4
0           3
Name: count, Length: 103, dtype: int64

In [16]:
clave_siianh = df_INAH_visitas['clave_siinah']
for clave in clave_siianh:
    if len(str(clave)) < 7:
        print(clave)

0
0
0


In [18]:
nombre_museo = df_INAH_visitas['recinto'].value_counts()
nombre_museo

<bound method IndexOpsMixin.value_counts of 0            Museo Regional de Historia de Aguascalientes
1       Museo Histórico Regional, Ex Cuartel de la Com...
2       Museo Regional de Antropología e Historia de B...
3                        Museo de Las Misiones Jesuíticas
4          Museo de las Pinturas Rupestres de San Ignacio
                              ...                        
3003    Museo de Sitio de Tres Zapotes con Zona Arqueo...
3004                                   Museo de Palmillas
3005           Museo Regional de Yucatán "Palacio Cantón"
3006           Pinacoteca del Estado "Juan Gamboa Guzmán"
3007                                   Museo de Guadalupe
Name: recinto, Length: 3008, dtype: str>

In [20]:
registros_clave_siinah_0 = df_INAH_visitas[df_INAH_visitas['clave_siinah'] == 0]
registros_clave_siinah_0

,anio,estado,clave_siinah,recinto,enero_nac,enero_ext,febrero_nac,febrero_ext,marzo_nac,marzo_ext,abril_nac,abril_ext,mayo_nac,mayo_ext,junio_nac,junio_ext,julio_nac,julio_ext,agosto_nac,agosto_ext,septiembre_nac,septiembre_ext,octubre_nac,octubre_ext,noviembre_nac,noviembre_ext,diciembre_nac,diciembre_ext
86,2025,Quintana Roo,0,Museo de la Costa Oriental,7452,3461,8562,5749,6942,3625,6942,3625,6804,2581,5283,2020,10214,3407,10276,3472,6198,2194,6711,2921,6635,3704,7325,3697
101,2025,Yucatán,0,Gran Museo de Chichén Itzá,392,310,442,229,532,324,389,208,330,173,258,161,384,174,523,286,288,75,409,217,684,472,461,399
205,2024,Yucatán,0,Gran Museo de Chichén Itzá,0,0,0,0,1099,185,752,102,569,138,449,111,346,103,139,73,78,17,345,87,307,203,482,174


In [23]:
recinto_museo_costa_oriental = df_INAH_visitas[df_INAH_visitas['recinto'] == 'Museo de la Costa Oriental']
recinto_museo_costa_oriental

,anio,estado,clave_siinah,recinto,enero_nac,enero_ext,febrero_nac,febrero_ext,marzo_nac,marzo_ext,abril_nac,abril_ext,mayo_nac,mayo_ext,junio_nac,junio_ext,julio_nac,julio_ext,agosto_nac,agosto_ext,septiembre_nac,septiembre_ext,octubre_nac,octubre_ext,noviembre_nac,noviembre_ext,diciembre_nac,diciembre_ext
86,2025,Quintana Roo,0,Museo de la Costa Oriental,7452,3461,8562,5749,6942,3625,6942,3625,6804,2581,5283,2020,10214,3407,10276,3472,6198,2194,6711,2921,6635,3704,7325,3697


In [28]:
museos_quintana_roo = df_INAH_visitas[df_INAH_visitas['estado'] == 'Quintana Roo']
museos_quintana_roo['recinto'].value_counts()

recinto
Museo Maya de Cancún con Z. A    24
Museo de la Costa Oriental        1
Name: count, dtype: int64

In [25]:
recinto_gran_museo_chichenitza = df_INAH_visitas[df_INAH_visitas['recinto'] == 'Gran Museo de Chichén Itzá']
recinto_gran_museo_chichenitza

,anio,estado,clave_siinah,recinto,enero_nac,enero_ext,febrero_nac,febrero_ext,marzo_nac,marzo_ext,abril_nac,abril_ext,mayo_nac,mayo_ext,junio_nac,junio_ext,julio_nac,julio_ext,agosto_nac,agosto_ext,septiembre_nac,septiembre_ext,octubre_nac,octubre_ext,noviembre_nac,noviembre_ext,diciembre_nac,diciembre_ext
101,2025,Yucatán,0,Gran Museo de Chichén Itzá,392,310,442,229,532,324,389,208,330,173,258,161,384,174,523,286,288,75,409,217,684,472,461,399
205,2024,Yucatán,0,Gran Museo de Chichén Itzá,0,0,0,0,1099,185,752,102,569,138,449,111,346,103,139,73,78,17,345,87,307,203,482,174


In [27]:
museos_yucatan = df_INAH_visitas[df_INAH_visitas['estado'] == 'Yucatán']
museos_yucatan['recinto'].value_counts()

recinto
Museo Regional de Yucatán "Palacio Cantón"    30
Pinacoteca del Estado "Juan Gamboa Guzmán"    30
Gran Museo de Chichén Itzá                     2
Museo de Sitio Ateneo Peninsular               1
Name: count, dtype: int64

In [30]:
museo_x = df_INAH_visitas[df_INAH_visitas['clave_siinah'] == 4100016]
museo_x

,anio,estado,clave_siinah,recinto,enero_nac,enero_ext,febrero_nac,febrero_ext,marzo_nac,marzo_ext,abril_nac,abril_ext,mayo_nac,mayo_ext,junio_nac,junio_ext,julio_nac,julio_ext,agosto_nac,agosto_ext,septiembre_nac,septiembre_ext,octubre_nac,octubre_ext,noviembre_nac,noviembre_ext,diciembre_nac,diciembre_ext
5,2025,Baja California Sur,4100016,Misión de San Francisco Javier,456,593,825,664,838,912,1081,568,818,245,616,259,762,211,1224,241,805,151,671,366,881,368,1347,247
83,2025,Puebla,4100016,Museo de sitio de Teteles del Santo Nombre con...,763,4,239,0,388,0,489,1,321,7,435,2,712,0,546,1,309,0,289,0,367,0,421,1
109,2024,Baja California Sur,4100016,Misión de San Francisco Javier,1486,663,1120,0,904,224,1250,110,406,107,960,167,1382,302,1335,154,516,176,698,372,823,838,884,636
188,2024,Puebla,4100016,Museo de sitio de Teteles del Santo Nombre con...,545,15,381,4,652,0,451,5,303,0,260,0,592,0,482,1,300,0,629,0,359,0,558,9
212,2023,Baja California Sur,4100016,Misión de San Francisco Javier,1546,1179,894,1352,820,1150,1615,887,637,269,507,72,1218,285,1965,221,774,150,733,266,807,275,3794,375
290,2023,Puebla,4100016,Museo de sitio de Teteles del Santo Nombre con...,826,4,642,0,535,0,696,0,405,0,677,0,522,0,458,0,557,0,645,0,644,0,711,15
313,2022,Baja California Sur,4100016,Misión de San Francisco Javier,1261,568,1143,1060,974,1010,1194,412,1042,535,496,135,1114,273,1254,243,1114,273,708,215,954,542,3172,368
391,2022,Puebla,4100016,Museo de sitio de Teteles del Santo Nombre con...,0,0,0,0,615,0,597,8,315,0,523,0,618,7,436,3,327,0,515,0,701,0,599,0
414,2021,Baja California Sur,4100016,Misión de San Francisco Javier,0,0,0,0,303,118,384,43,409,123,486,176,0,0,0,0,0,0,0,0,217,37,1466,320
492,2021,Puebla,4100016,Museo de sitio de Teteles del Santo Nombre con...,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0


1. No se identificaron valores faltantes o valores incorrectos en las columnas anios, estados, recintos.
2. Se identificaron 3 calves siinah de un solo digito cuando las demás claves son 7 digitos.
3. Los valores 0 en las columnas de visitas nacionales y extranjeros no son errores, representan total ausencia de visitas por causas como cierre del museo, remodelaciones, cero interes en las excibiciones e inclusive por inseguridad.
4.-Las claves siinah no son identificadores unicos de museso, por lo que son irrelevantes.